Linkcheck plugin, excluded broken links until corresponding plugins will...

alexkravin opened the pull request:
... be fixed, issue #549

According to #549 

Detailed analyze of the problem in links determined that most of problems are not on our side (such problems are already fixed), so due to these problems I have opened 3 issues on:

1) maven-javadoc-plugin, which uses outdated doxia's version: http://jira.codehaus.org/browse/MJAVADOC-425
2) maven-linkcheck-plugin: http://jira.codehaus.org/browse/MPLINKCHECK-32
3) doxia, which renders links in velocity templates very weird (for example it break parentheses): http://jira.codehaus.org/browse/DOXIA-525

Until these issues will be resolved - corresponding links are excluded, after updating of plugins and testing they will be returned on their places.

List of excluded links:

1) Links to javadoc methods in apidocs (spaces in links are not supported by linkcheck-plugin)
2) http://docs.oracle.com/javase/7/docs/api/org/xml/sax/helpers/DefaultHandler.html?is-external=true
from AbtractLoader class which is mostly used in filters, maven-javadoc-plugin generates links on <b>filters</b> package, e.g. http://alexkravin.github.io/linkcheck/apidocs/com/puppycrawl/tools/checkstyle/filters/package-tree.html
and specifies links from inherited classes too, as you can see link to Default handler is broken
http://docs.oracle.com/javase/7/docs/api/org/xml/sax/helpers<b>.</b>DefaultHandler.html?is-external=true

all other links to classes at docs.oracle/... are ok.

3) Links to xref-sources. We use the last version of jxr plugin (2.4), but links are broken, e.g.:
./xref/com/puppycrawl/tools/checkstyle/api/FullIdent.html#91
should be
./xref/com/puppycrawl/tools/checkstyle/api/FullIdent.html#<b>L</b>91

The same problem we can see here http://maven.apache.org/plugins/maven-checkstyle-plugin/taglist.html
Just click to any line number - it won't redirect you to proper line in xref-sources. I think this problem is in scope of maven-linkcheck-plugin too

Current Linkcheck report:
http://alexkravin.github.io/currentlinks/linkcheck.html
there're some errors, but all of them are of 2 types:
1) Timeout 2000ms (my connection is not fast enough)
2) 403 forbidden - that's because of my location.

romani commented:
1)
> http://jira.codehaus.org/browse/DOXIA-525

Please never give links to web sites that show issue tracker as mail-list, always provide link to original source.
https://jira.codehaus.org/browse/DOXIA-397 is fixed, so problem should be resolved on our side.

2) 
> DefaultHandler.html

I already told you few times , you need to prove that on simple project and report that issue to their issue tracker.

3) 
>  Links to xref-sources. We use the last version of jxr plugin (2.4)

Please prove by effective-pom that there is no old plugin. If issue still exists please report to linkcheck issue tracker.

4) One more time, against each exclude tag you should put issue number (in a comment) that a reason of it. That will let easily see that numbers during pom.xml review and recheck in web for status.

alexkravin commented:
1) Problem fixed here https://jira.codehaus.org/browse/DOXIA-397 was the problem if link is specified in javadoc, we have problem if link is specified in velocity template, we can't escape link with double '##' as commented in issue, because in that case Doxia will fail:
```
[ERROR] Error parsing /home/asus/sources/links/src/xdocs/writingchecks.xml.vm as a velocity template, using as text.
```
that's why I've created separate issue against this.

2) My test project https://github.com/alexkravin/linkstest
it's linkcheck report
http://alexkravin.github.io/testproject/linkcheck.html
Extended issue against linkcheck-plugin

3) In effective pom jxr-plugin is occurred 5 times:

4 times as
```xml
<reportPlugin>
                  <groupId>org.apache.maven.plugins</groupId>
                  <artifactId>maven-jxr-plugin</artifactId>
                  <version>2.4</version>
</reportPlugin>
```
and 1 time as
```xml
<plugin>
        <artifactId>maven-jxr-plugin</artifactId>
        <version>2.4</version>
</plugin>
```

So, we use proper version, description of this problem was already written in the end of my issue against linkcheck-plugin

4) Done

alexkravin commented:
Travis broke the connection again

romani commented:
1) for javadoc generation old doxia is used - ok, What version of Doxia is used during site genetration by plugin? are you sure that we use 1.6 version of Doxia during site generation?

2)
> In effective pom jxr-plugin is occurred 5 times

the latest version is 2.5 (not a 2.4). Can we force site-plugin to use new version of jxr ?

3) 
Why you reported "http://docs.oracle.com/javase/7/docs/api/org/xml/sax/helpers.DefaultHandler.html?is-external=true: 404 Not Found" to http://jira.codehaus.org/browse/MLINKCHECK-21, ? 
linkcheck is not resposible for links generation.


call me in skype to discuss that issue.

alexkravin commented:
1) Yes, I'm sure about site-plugin, as we use version 3.4, its pom fragments:
```xml
<dependency>
      <groupId>org.apache.maven.doxia</groupId>
      <artifactId>doxia-sink-api</artifactId>
      <version>1.6</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>org.apache.maven.doxia</groupId>
      <artifactId>doxia-logging-api</artifactId>
      <version>1.6</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>org.apache.maven.doxia</groupId>
      <artifactId>doxia-core</artifactId>
      <version>1.6</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>org.apache.maven.doxia</groupId>
      <artifactId>doxia-module-xhtml</artifactId>
      <version>1.6</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>org.apache.maven.doxia</groupId>
      <artifactId>doxia-module-apt</artifactId>
      <version>1.6</version>
      <scope>runtime</scope>
    </dependency>
    <dependency>
      <groupId>org.apache.maven.doxia</groupId>
      <artifactId>doxia-module-xdoc</artifactId>
      <version>1.6</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>org.apache.maven.doxia</groupId>
      <artifactId>doxia-module-fml</artifactId>
      <version>1.6</version>
      <scope>runtime</scope>
    </dependency>
    <dependency>
      <groupId>org.apache.maven.doxia</groupId>
      <artifactId>doxia-module-markdown</artifactId>
      <version>1.6</version>
      <scope>runtime</scope>
    </dependency>
...
<properties>
...
    <doxiaSitetoolsVersion>1.6</doxiaSitetoolsVersion>
...
    <doxiaVersion>1.6</doxiaVersion>
```

2) It's not the problem of site plugin, it's problem of maven-linkcheck-plugin which uses doxia 1.4, which uses jxr version 2.3.
Do you remember the first time I was generating report against some projects? I used jxr 2.3 too in config and has the same problem with lack of literal <b>L</b> in the links.

3) You're right, it's the problem of maven-javadoc-plugin too, removed this part from issue against linkcheck-plugin, extended issue against maven-javadoc-plugin, changed link to bug in pom.xml (above excluded link)
